src/Symfony/Component/Form/CHANGELOG.md

```diff
@@ -1,6 +1,12 @@
 CHANGELOG
 =========
 
+2.7.0
+-----
+
+ * deprecated AbstractType::setDefaultOptions() in favor of AbstractType::configureOptions().
```

Comment:
no. the method itself is not deprecated (it is not possible as it **must** be implemented in form types and is being used.
What is deprecated is overwriting ``AbstractType::setDefaultOptions()``, in favor of overwriting ``AbstractType::configureOptions()``, to use a forward-compatibility layer.

This is not a BC layer here. We are making it possible to write your code as if you were using the Symfony 3 interface already, even though it is not the case yet, handling the differences in the parent class. This is not the same than providing a new API and deprecating the BC layer